data-structures/ruby/min_stack.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'stack'
+
+class MinStack < Stack
+
+  def initialize
+    super
+    @min_stack = Stack.new()
+  end
+
+  def min
+    @min_stack.peek
+  end
+
+  def push(value)
+    @min_stack.push(value) if @min_stack.empty? || @min_stack.peek <= value
+    super(value)
+  end
+
+  def pop
+    @max_stack.pop if peek == @max_stack.peek
+    super
+  end
```

Comment:
Hmm, this works — I think!  Consider using a single `Stack` and pushing a tuple onto it instead of a single value.  The first element of the tuple would be the value and the second element would be the largest thing in the stack at that point.

```ruby
def push(value)
  super([value, [value, max].max])
end
```

You might need to rejigger it a bit to get all the methods to work, but that's the main idea.